package.json

```diff
@@ -8,7 +8,9 @@
     "clean": "rm -rf lib",
     "build": "npm run clean && tsc",
     "generate": "./generate.sh",
-    "test": "mocha --recursive -r ts-node/register -r source-map-support/register 'test/**/*.ts'",
+    "test": "npm run test:unit && npm run test:integration",
```

Comment:
@easyCZ this is the bulk of the change which consists of 2 parts:

1. Change the layout of the tests on disk making a clear split between integration and unit tests. This allowed me to remove the `test/test_ts` folder which contained a single unit test, some helper code (used by the integration tests) and some scripts for asserting the output from mocha (see point 2 below).

2. Invoke the new `mocha-run-suite.sh` script which invokes mocha and then performs a quick assertion to check that at least 1 test passed. Without this check it's possible for zero tests to run but mocha to still return exit code 0 marking the build as green when it shouldn't be.